ergotree-interpreter/src/eval/xor_of.rs

```diff
@@ -0,0 +1,40 @@
+use ergotree_ir::mir::constant::TryExtractInto;
+use ergotree_ir::mir::value::Value;
+
+use crate::eval::env::Env;
+use crate::eval::EvalContext;
+use crate::eval::EvalError;
+use crate::eval::Evaluable;
+use ergotree_ir::mir::xor_of::XorOf;
+
+impl Evaluable for XorOf {
+    fn eval(&self, env: &Env, ctx: &mut EvalContext) -> Result<Value, EvalError> {
+        let input_v = self.input.eval(env, ctx)?;
+        let input_v_bools = input_v.try_extract_into::<Vec<bool>>()?;
+        Ok((input_v_bools.into_iter().filter(|x| *x).count() & 1 == 1).into())
+    }
+}
+
+#[allow(clippy::panic)]
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::eval::context::Context;
+    use crate::eval::tests::eval_out;
+    use ergotree_ir::mir::expr::Expr;
+    use proptest::collection;
+    use proptest::prelude::*;
+    use sigma_test_util::force_any_val;
+    use std::rc::Rc;
+
+    proptest! {
+
+        #[test]
+        fn eval(bools in collection::vec(any::<bool>(), 10..=20)) {
+            let expr: Expr = XorOf {input: Expr::Const(bools.clone().into()).into()}.into();
+            let ctx = Rc::new(force_any_val::<Context>());
+            let res = eval_out::<bool>(&expr, ctx);
+            prop_assert_eq!(res, bools.into_iter().filter(|x| *x).count() & 1 == 1);
```

Comment:
There is not much sense in testing it's semantic equivalence with itself. Let's use a different algo here, whatever you prefer. Also, please test on empty collection as well (extend range to 0..=20).

Reply:
I updated the test with new range (forgot to revert after doing some personal testing) and also to use a mutable val to hold the reassigned result after each XOR gate.  To your point, this solution is likely more readable than the eval function, so if you prefer maybe I swap them?

Reply:
Yes, please swap it. I find XOR gate with mutable accum more readable.